### PR TITLE
Add systemuser validator

### PR DESCRIPTION
**Pull request by chenwany**
…e name

Signed-off-by: chenwany <chenwany@amazon.com>

### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -1027,7 +1027,9 @@ def make_resource(self, data, **kwargs):
 class _ComputeResourceSchema(BaseSchema):
     """Represent the schema of the ComputeResource."""
 
-    name = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    name = fields.Str(
```
Comment (lukeseawalker) [inline cli/src/pcluster/schemas/cluster_schema.py]: name validation is already done at config level in `NameValidator`, please remove this

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 2]
```diff
@@ -1098,7 +1100,9 @@ def make_resource(self, data, **kwargs):
 class BaseQueueSchema(BaseSchema):
     """Represent the schema of the attributes in common between all the schedulers queues."""
 
-    name = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    name = fields.Str(
```
Comment (lukeseawalker) [inline cli/src/pcluster/schemas/cluster_schema.py]: same here, already done in config

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 3]
```diff
@@ -1441,7 +1445,9 @@ def make_resource(self, data, **kwargs):
 class SchedulerPluginUserSchema(BaseSchema):
     """Represent the schema of the Scheduler Plugin."""
 
-    name = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    name = fields.Str(
```
Comment (lukeseawalker) [inline cli/src/pcluster/schemas/cluster_schema.py]: I think it's better to have a config validator here and not only avoid the new lines, but validate the name as Linux user name. You can use the same regex used here https://github.com/aws/aws-parallelcluster-cookbook/blob/4756da1f9660bcfb94cd8f59a7e5b3e8886e67b4/cookbooks/aws-parallelcluster-config/files/default/dcv/pcluster_dcv_authenticator.py#L130

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 4]
```diff
@@ -1727,7 +1733,11 @@ def restore_child(self, data, **kwargs):
 class DirectoryServiceSchema(BaseSchema):
     """Represent the schema of the DirectoryService."""
 
-    domain_name = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.COMPUTE_FLEET_STOP})
+    domain_name = fields.Str(
```
Comment (lukeseawalker) [inline cli/src/pcluster/schemas/cluster_schema.py]: same here, we can do a better validation for this string. But I think we can leave this out of this PR and track in a different task

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 1 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Removed

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 2 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Removed

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 3 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Added

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 4 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Removed

**Review by lukeseawalker — DISMISSED** [review]

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**